zetkin/app.zetkin.org
# Page for opening a journey

## Description
This PR creates a page where a new journey instance can be opened.

## Screenshots
https://user-images.githubusercontent.com/550212/167681554-494c9e7a-4f82-4ccd-9478-2f19b5bff364.mov

## Change
* Adds a new page where journey instances can be opened from
* Creates NEXT API route used to open a journey instance against the Zetkin API, while adding assignees and subjects
* Also a number of changes to existing code to support this:
  * Refactors layouts to use use common components
  * Adds placeholder to `EditTextInPlace`
  * Breaks out `ZetkinAutoTextArea` into it's own component

## Notes to reviewer

### Deviations from design
* The FAB on the journey instance list page is at the bottom of the screen, contrary to the designs. The designs do not include the "collapse" option in the header, which complicates having the FAB there. Instead, I've put it in the bottom right, above the table footer
* On the new page, there were a lot of elements in the designs that do not make sense on this page where the journey has not yet been started, e.g. ellipsis menu, "last updated" label, tabs and milestones
* The tags have not been implemented yet, and are tracked separately in #578 

### How to test
1. Go to cases list, http://localhost:3000/organize/1/journeys/1
2. Click FAB navigate to new page
3. Add title and opening note text
4. Optionally add an assignee
5. Optionally add a member
6. Click create button

A new case should be opened and you should be taken to it's instance page.

## Related issues
Resolves #580
